department-of-veterans-affairs/LEAF
# LEAF 4455 - reading metadata for form requests (orgchart employee format)

Updates orgchart_employee format form questions and report builder cells to use saved metadata from the data table rather than calling lookupEmpUID.

Adds the metadata field values to the API-testing portal db initial boilerplate (this db is otherwise already up to date with table structure).

other: script to update data.metadata added to leaf-scripts.  This is potentially needed for local/dev envs with older volumes of leaf_portal data.  Orgchart employee entries without metadata would otherwise show as disabled accounts.  Pending some further discussion.

Finish associated Go test


**Impact / Testing**

Impact is to orgchart employee format questions on form requests and Report Builder reports.  Impact is high and could cause data for orgchart_employee format questions to display incorrectly in form requests and Report Build reports.

- Confirm orgchart employee format entries display correctly in form requests and in Report Builder reports.

- Run Go tests on branch **issue/LEAF-4478/orgchart_info_go_test** as well as on this one and confirm they pass.  On 4478, orgchart values come from a query of the orgchart, on 4455, they come from the metadata field.

- Confirm employee entries with metadata continue to display if they are subsequently disabled.
Note: entries associated with accounts that were **already disabled** when the metadata update script was run should continue to display disabled account information.
